torch/lib/TensorLib/Context.h

```diff
@@ -0,0 +1,58 @@
+#pragma once
+
+#include <memory>
+#include "TensorLib/Generator.h"
+#include "TensorLib/Type.h"
+
+class THCState;
+
+namespace tlib {
+
+class Context {
+public:
+  Context();
+  Type & getType(Processor p, ScalarType s) {
+    auto & type = type_registry[static_cast<int>(p)][static_cast<int>(s)];
+    if(!type)
+      throw std::runtime_error("Type %s is not enabled (TODO encode type as string)");
```

Comment:
The TODO is there because we didnt implement the Processor or ScalarType enum to string methods so that this error can be better...